internal/controller/service_accounts.go

```diff
@@ -16,14 +17,33 @@ const (
 	modelServerServiceAccountName      = "model-server"
 	notebookServiceAccountName         = "notebook"
 	dataLoaderServiceAccountName       = "data-loader"
+	identityBoundLabel                 = "substratusai/identity-bound"
 )
 
-func reconcileCloudServiceAccount(ctx context.Context, cloudConfig cloud.Cloud, c client.Client, sa *corev1.ServiceAccount) (result, error) {
+func reconcileCloudServiceAccount(ctx context.Context, cloudConfig cloud.Cloud, sciClient sci.ControllerClient, c client.Client, sa *corev1.ServiceAccount) (result, error) {
```

Comment:
Aligning with naming elsewhere:
```suggestion
func reconcileCloudPrincipal(ctx context.Context, cloudConfig cloud.Cloud, sciClient sci.ControllerClient, c client.Client, sa *corev1.ServiceAccount) (result, error) {
```

Reply:
I think we should rename to reconcileServiceAccount

Reply:
We reconcile the service account here (create/update the KSA)... as a part of doing that we reconcile the cloud principle. I like `reconcileServiceAccount` as the func name.

Reply:
Ahh yeah `reconcileServiceAccount` works best here

Reply:
done, renamed to `reconcileServiceAccount`